application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml

```diff
@@ -59,13 +60,24 @@
       #set($list = $obj.getProperty('flash').getValue())
       #if($list == $util.getNull())
         #set($discard = $obj.set('flash', [$xcontext.user]))
+        #set($discard = $obj.set('seenDate', $dateNow))
         #set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
       #elseif($list.class.name == 'com.xpn.xwiki.objects.ListProperty$NotifyList' &amp;&amp; !$list.contains($xcontext.user))
         #set($discard = $list.add($xcontext.user))
         #set($discard = $obj.set('flash', $list))
+        #set($discard = $obj.set('seenDate', $dateNow))
```

Comment:
This line is duplicated few lines above. You can set the `seenDate` property outside the `if-else` statement. I see that the same happens with `#set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))`.